src/zenml/integrations/spark/step_operators/spark_step_operator.py

```diff
@@ -0,0 +1,325 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+import json
+import subprocess
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from pydantic import validator
+from pyspark.conf import SparkConf
+
+from zenml.config.docker_configuration import DockerConfiguration
+from zenml.logger import get_logger
+from zenml.repository import Repository
+from zenml.step_operators import BaseStepOperator, entrypoint
+
+logger = get_logger(__name__)
+if TYPE_CHECKING:
+    from zenml.config.resource_configuration import ResourceConfiguration
+
+LOCAL_ENTRYPOINT = entrypoint.__file__
+APP_DIR = "/app/"
+CONTAINER_ZENML_CONFIG_DIR = ".zenconfig"
+ENTRYPOINT_NAME = "zenml_spark_entrypoint.py"
+
+
+class SparkStepOperator(BaseStepOperator):
+    """
+    MAIN DOCS
+    """
+
+    # Instance parameters
+    master: str
+    deploy_mode: str = "cluster"
+
+    # Additional parameters for the spark submit
+    submit_kwargs: Optional[Dict[str, Any]] = None
+
+    @property
+    def application_path(self) -> Any:
+        """Optional method for providing the application path.
+
+        This is especially critical when using 'spark-submit' as it defines the
+        path (to the application in the environment where Spark is running)
+        which is used within the command.
+
+        For more information on how to set this property please check:
+
+        https://spark.apache.org/docs/latest/submitting-applications.html#advanced-dependency-management
+        """
+        return None
```

Comment:
This is a property that always returns None? What am I missing, why can't this be an optional instance parameter?

Reply:
The application path is a property that _might_ be used within the `_launch_spark_job` method depending on the case. This is mainly the reason why I did not want to turn it into an instance parameter.

In cases such as the `KubernetesSparkStepOperator`, it points to where the entrypoint for the step lies (for the `spark-submit` command). This way, if our users decide to use a different base image (with a different application path), they can just change this property without changing the `_launch_spark_job` method.

Reply:
Does it make sense to at least make it return `Optional[str]` in this case?

Reply:
Makes sense. Fixed it.